zebra-network/src/peer_observer.rs

```diff
@@ -0,0 +1,17 @@
+//! A PeerObserver trait for getting the [`MetaAddr`] of recently live peers.
+
+use chrono::Utc;
+
+use crate::meta_addr::MetaAddr;
+
+#[cfg(any(test, feature = "proptest-impl"))]
+pub mod mock;
+
+#[cfg(any(test, feature = "proptest-impl"))]
+pub use mock::MockPeerObserver;
+
+/// Method signatures for getting [`MetaAddr`]s of recently live peers.
+pub trait PeerObserver {
```

Comment:
Any suggestions for this trait name?

Reply:
Usually Rust names single-method traits after the method name.

But since we're likely to add more methods later, I'd go with something like:
* `AddressBookPeers`
* `AddressBookAccess`
* `AddressBookContents`

I'd avoid anything that suggests a notification, like `Change` or `Observer`. We already have traits that notify on changes in Zebra, so that might be confusing.

Reply:
Thank you, that sounds better. Renamed in [`4bf183e`](https://github.com/ZcashFoundation/zebra/pull/5951/commits/4bf183eafbc1ed300385d8b8724c16f6a7543bc1).